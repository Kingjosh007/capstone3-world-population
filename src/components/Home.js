import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Topbar from './Topbar';
import { getCountriesInfos } from '../redux/countries/countries';
import Hero from './Hero';
import DisplayArea from './DisplayArea';

const Home = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getCountriesInfos());
  }, [dispatch]);

  const countriesArr = useSelector((state) => state.countriesReducer.countries);
  let worldInfos = countriesArr.find((ctr) => ctr.name === 'World');
  if (countriesArr.length === 0) {
    worldInfos = {
      name: 'Unknown',
      flag: 'https://upload.wikimedia.org/wikipedia/commons/8/87/Flag-map_of_the_world.svg',
      latestPop: 'Unknown',
      latestPopYear: '2022',
    };
  }

  const {
    name, flag, latestPop, latestPopYear,
  } = worldInfos;

  return (
    <div>
      <Topbar title="World population" />
      <Hero region={name} population={latestPop} year={latestPopYear} imgLink={flag} />
      <DisplayArea type="countries" title="Population by country" />
    </div>
  );
};

export default Home;
