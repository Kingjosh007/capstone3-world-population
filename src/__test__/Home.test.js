import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('Home component is displayed properly', () => {
  test('renders Home', () => {
    const tree = render(
      <Provider store={store}>
        <React.StrictMode>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </React.StrictMode>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
