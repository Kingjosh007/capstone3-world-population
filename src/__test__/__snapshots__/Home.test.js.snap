// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component is displayed properly renders Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="topbar primary-3"
        >
          <div>
            <a
              class="text-1"
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                />
              </svg>
            </a>
          </div>
          <div>
            <h5
              class="main-title text-2"
            >
              World population
            </h5>
          </div>
          <div
            class="icons-right"
          >
            <svg
              class="text-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 352 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
              />
            </svg>
            <svg
              class="text-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
              />
            </svg>
          </div>
        </nav>
        <div
          class="hero primary-4"
        >
          <div
            class="hero-image"
          >
            <img
              alt="Flag or area of Unknown"
              src="https://upload.wikimedia.org/wikipedia/commons/8/87/Flag-map_of_the_world.svg"
            />
          </div>
          <div
            class="hero-content"
          >
            <div
              class="hero-region"
            >
              Unknown
            </div>
            <div
              class="hero-pop"
            >
              <div>
                <span
                  class="hero-actual-pop"
                >
                  Unknown
                </span>
                 inhab.
              </div>
              <div
                class="hero-year"
              >
                (
                2022
                )
              </div>
            </div>
          </div>
        </div>
        <div
          class="display-area"
        >
          <div
            class="display-header primary-3"
          >
            <div
              class="display-header-left"
            >
              <h3
                class="text-2"
              >
                Population by country
              </h3>
              <div
                class="hint"
              />
            </div>
            <div
              class="filter-input"
            >
              <input
                class="input col sm-6 md-6 is-valid"
                id="some-input"
                name="some-input"
                placeholder="Search countries"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="countries-container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="topbar primary-3"
      >
        <div>
          <a
            class="text-1"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 256 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
              />
            </svg>
          </a>
        </div>
        <div>
          <h5
            class="main-title text-2"
          >
            World population
          </h5>
        </div>
        <div
          class="icons-right"
        >
          <svg
            class="text-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 352 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
            />
          </svg>
          <svg
            class="text-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
            />
          </svg>
        </div>
      </nav>
      <div
        class="hero primary-4"
      >
        <div
          class="hero-image"
        >
          <img
            alt="Flag or area of Unknown"
            src="https://upload.wikimedia.org/wikipedia/commons/8/87/Flag-map_of_the_world.svg"
          />
        </div>
        <div
          class="hero-content"
        >
          <div
            class="hero-region"
          >
            Unknown
          </div>
          <div
            class="hero-pop"
          >
            <div>
              <span
                class="hero-actual-pop"
              >
                Unknown
              </span>
               inhab.
            </div>
            <div
              class="hero-year"
            >
              (
              2022
              )
            </div>
          </div>
        </div>
      </div>
      <div
        class="display-area"
      >
        <div
          class="display-header primary-3"
        >
          <div
            class="display-header-left"
          >
            <h3
              class="text-2"
            >
              Population by country
            </h3>
            <div
              class="hint"
            />
          </div>
          <div
            class="filter-input"
          >
            <input
              class="input col sm-6 md-6 is-valid"
              id="some-input"
              name="some-input"
              placeholder="Search countries"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="countries-container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
